common/djangoapps/track/tests/__init__.py

```diff
@@ -87,3 +88,71 @@ def assert_events_emitted(self):
     def assertEqualUnicode(self, tree_a, tree_b):
         """Like assertEqual, but give nicer errors for unicode vs. non-unicode"""
         self.assertEqual(unicode_flatten(tree_a), unicode_flatten(tree_b))
+
+    def assert_event_matches(self, expected, actual, strict=False):
+        """
+        Compare two event dictionaries.
+
+        Fail if any discrepancies exist, and output the list of all discrepancies. The intent is to produce clearer
+        error messages than "{ some massive dict } != { some other massive dict }", instead enumerating the keys that
+        differ. Produces period separated "paths" to keys in the output, so "context.foo" refers to the following
+        structure:
+
+            {
+                'context': {
+                    'foo': 'bar'  # this key, value pair
+                }
+            }
+
+        By default, it only asserts that all fields specified in the first event also exist and have the same value in
+        the second event. If the `strict` parameter is passed in it will also ensure that *only* the fields in the first
+        event exist in the second event. For example::
+
+            expected = {
+                'a': 'b'
+            }
+
+            actual = {
+                'a': 'b',
+                'c': 'd'
+            }
+
+            self.assert_events_equal(expected, actual, strict=False)  # This will not raise an AssertionError
+            self.assert_events_equal(expected, actual, strict=True)   # This *will* raise an AssertionError
+        """
+        errors = self._compare_trees(expected, actual, strict, [])
+        if len(errors) > 0:
+            self.fail('Unexpected event differences found:\n' + '\n'.join(errors))
+
+    def _compare_trees(self, expected, actual, strict, path):
+        errors = []
+
+        if not strict and path == ['event'] and isinstance(expected, dict) and isinstance(actual, basestring):
```

Comment:
When the time comes, it might be good to add a comment here indicating why we parse the json in this case.